cmd/argo/commands/template/lint.go

```diff
@@ -1,122 +1,48 @@
 package template
 
 import (
-	"context"
 	"fmt"
 	"os"
-	"path/filepath"
 
-	log "github.com/sirupsen/logrus"
+	"github.com/argoproj/pkg/errors"
 	"github.com/spf13/cobra"
 
 	"github.com/argoproj/argo-workflows/v3/cmd/argo/commands/client"
-	workflowtemplatepkg "github.com/argoproj/argo-workflows/v3/pkg/apiclient/workflowtemplate"
-	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
-	cmdutil "github.com/argoproj/argo-workflows/v3/util/cmd"
-	"github.com/argoproj/argo-workflows/v3/workflow/validate"
+	"github.com/argoproj/argo-workflows/v3/workflow/lint"
 )
 
 func NewLintCommand() *cobra.Command {
 	var (
 		strict bool
+		format string
 	)
 	var command = &cobra.Command{
 		Use:   "lint (DIRECTORY | FILE1 FILE2 FILE3...)",
 		Short: "validate a file or directory of workflow template manifests",
 		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) == 0 {
-				cmd.HelpFunc()(cmd, args)
+			ctx, apiClient := client.NewAPIClient()
+			fmtr, err := lint.GetFormatter(format)
+			errors.CheckError(err)
+
+			res, err := lint.Lint(ctx, &lint.LintOptions{
+				ServiceClients: lint.ServiceClients{
+					WorkflowTemplatesClient: apiClient.NewWorkflowTemplateServiceClient(),
+				},
+				Files:            args,
+				Strict:           strict,
+				DefaultNamespace: client.Namespace(),
+				Formatter:        fmtr,
+			})
+			errors.CheckError(err)
+
+			fmt.Print(res)
```

Comment:
Can we do this instead? More explicit
```suggestion
			fmt.Print(res.Msg)
```

Reply:
I could do that. I thought it's cleaner, since I made `LintResults` type implement the `Stringer` interface. This way it's also readonly. But I guess I could just add a `.Msg()` method that returns the internal message instead.